<?xml version="1.0"?>

<launch>


  <!-- ROS serial node -->
  <node pkg="rosserial_python" type="serial_node.py" name="serial_node">

    <!-- Name of port to use (Teensy)-->
    <param name="port" value="/dev/ttyACM0"/>

    <!-- Baud rate, in bps -->
    <param name="baud" value="57600"/>

  </node>


  <!-- Kill node -->
  <node pkg="race2" type="kill.py" name="kill" output="screen"
   launch-prefix="xterm -e"/>



  <!-- Talker node -->
  <node pkg="race2" type="talker.py" name="talker" output="screen" 
   launch-prefix="xterm -e"/>



  <!-- Control node -->
  <node pkg="race2" type="control.py" name="control" output="screen"
   launch-prefix="xterm -e">

    <param name="kp" value="14.0"/>
    <param name="kd" value="0.09"/>
    <param name="scale" value="7.0"/>

  </node>



  <!-- Distance finder node -->
  <node pkg="race2" type="path_dist_finder.py" name="path_dist_finder" output="screen"
   launch-prefix="xterm -e">

    <param name="velocity" value="9.0"/>

  </node>

  <!-- py_pose_flag -->
  <node pkg="race2" type="py_pose_flag.py" name="py_pose_flag" output = "screen"
   launch-prefix="xterm -e">

	<param name="x_stop" value="0.75"/>
	<param name="y_stop_min" value="-0.3"/>
	<param name="y_stop_max" value="0.3"/>

  </node>

  <!-- hm_amcl -->
  <include file="$(find race2)/launch/hm_amcl.launch"/>



</launch>
